README.md

```diff
@@ -8,6 +8,62 @@ a threshold.
 The ultimately use case for this script is to run it periodically and use it to
 trigger alerts/notifications that the Product Delivery team can take heed of.
 
+## Architectural Overview
+
+### In Plain English
+
+This is stateless script, written in Python, meant to be executed roughly once a day.
+It uses `requests.get()` to query [Redash](https://sql.telemetry.mozilla.org) for
+a query's data that is pre-emptively run every 24 hours. The data it analyses is a
+list of every collection stored in Remote Settings (Kinto) along with a count of
+every possible status (e.g. `up_to_date`, `network_error`, etc.)
+The script sums the "good" statuses and compares it against the "bad" statuses and
+if that ratio (expressed as a percentage) is over a certain threshold it alerts
+by sending an event to Sentry which notifies the team by email.
+
+### Historical Strategy
+
+As of Feb 2019, Firefox clients that use Remote Settings only send Telemetry events
+in daily batches. I.e. it's _not_ real-time. The "uptake histogram" is buffered in
+the browser and will send periodically instead of as soon as possible. In Firefox
+Nightly (67 at the time of writing), we are switching to real-time Telemetry Events.
+
+On the Telemetry backend we're still consuming the older uptake histogram but once,
+what is Nightly today, becomes Release, we will switch the Redash query (where
+appropriate) and still use this script to worry about the percentage thresholds.
+And the strategy for notifications should hopefully not change.
+
+Essentionally, once we switch to real-time Telemetry Events, nothing changes in
+terms of the surface API but the underlying data is more up-to-date and the response
+time of reacting to failure spikes is faster.
```

Comment:
We could also mention that in the mid/long term, based on the Telemetry tools available to do some alerting we could decomission this one.